obspy/core/frequency_domain_trace.py

```diff
@@ -59,77 +44,69 @@ def phase(self):
     def amplitude(self):
         return np.absolute(self.data)
     
-    def cross_correlation(self, tr2):
-
-        if self.stats.starttime != tr2.stats.starttime:
+    def cross_correlation(self, other):
+        if self.stats.starttime != other.stats.starttime:
             msg = "Traces do not have the same starttime"
             raise Exception(msg)
-       
-        elif "FALSE" == np.array_equal(self.frequencies(), tr2.frequencies()):
+        elif "FALSE" == np.array_equal(self.frequencies, other.frequencies):
```

Comment:
You're comparing a string with a boolean value, so this will never evaluate to `True`.
`np.array_equal` is returning a `bool` value.. `np.array_equal(...) is False`